examples/amp-story/bookendv2.json

```diff
@@ -0,0 +1,25 @@
+{
+  "bookend-version": "v2.0",
+  "components": [
+    {
+      "type": "share-providers",
```

Comment:
If we put this inline as a component, we need to figure out how to communicate:

(1) *Don't communicate; allow reordering:* I would assume this to be the default; wherever publishers put the `share-providers` component, it shows up.  This is problematic because we generally want to ensure that it comes first, that there is at most one, and that if unspecified, we add it as first anyway.
(2) *Don't communicate; disallow reordering:* We take the first `share-providers` instance we find and make it the first component.  If they specify more than one, we silently drop the rest (or somehow merge them).
(3) *Console error:* If they specify a `share-providers` component as anything other than the first component, we throw a console error and disregard that component entirely.  This provides flexibility, guarantees there is only one, and guarantees that it comes first, but is potentially confusing to publishers (e.g. if they don't look at the console)
(4) *Separate entity:* We can move the `share-providers` out of the `components` object (as it was before), and we insert it as the first thing at runtime.  This guarantees there is only one and that it is first, but is relatively unflexible.

Reply:
Hmm I'm leaning more towards moving it out of the `components` object, as it was before. It seems that with any approach that we take, we would enforce the position of the share-providers at some level. 

Having it outside of the `component` object seems to align with these semantics.